common/modules/utils/src/lib.rs

```diff
@@ -24,13 +26,35 @@ pub trait UtilsModule {
         )
     }
 
+    fn burn_multi_esdt(&self, payments: &PaymentsVec<Self::Api>) {
+        for payment in payments {
+            self.send().esdt_local_burn(
+                &payment.token_identifier,
+                payment.token_nonce,
+                &payment.amount,
+            );
+        }
+    }
+
     fn get_non_empty_payments(&self) -> PaymentsVec<Self::Api> {
         let payments = self.call_value().all_esdt_transfers();
-        require!(!payments.is_empty(), "No payments");
+        require!(!payments.is_empty(), ERR_EMPTY_PAYMENTS);
 
         payments
     }
 
+    fn pop_first_payment(
+        &self,
+        payments: &mut PaymentsVec<Self::Api>,
+    ) -> EsdtTokenPayment<Self::Api> {
+        require!(!payments.is_empty(), ERR_EMPTY_PAYMENTS);
```

Comment:
Why not use `get_non_empty_payments` here?

Reply:
Payments is an argument here, so it modifies a given payment Vec.